storage/redistore/peer_crud.go

```diff
@@ -0,0 +1,70 @@
+package redistore
+
+import (
+	"encoding/binary"
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/chihaya/chihaya/bittorrent"
+	"github.com/garyburd/redigo/redis"
+)
+
+func decodePeerKey(pk string) bittorrent.Peer {
+	return bittorrent.Peer{
+		ID:   bittorrent.PeerIDFromString(string(pk[:20])),
+		Port: binary.BigEndian.Uint16([]byte(pk[20:22])),
+		IP:   net.IP(pk[22:]),
+	}
+}
+
+func addAndCleanPeer(s *peerStore, infoHash bittorrent.InfoHash, peerType string, pk serializedPeer) error {
+	nameSpacedKey := addNameSpace(fmt.Sprintf("%s%s", peerType+":", infoHash))
```

Comment:
The peerType may vary for every format. Can I precompute the namespace into the constants Seeder and Leecher? That may remove the need for addNameSpace.